File: src/CellularSignalStrengthModule.ts
import { NativeModule, requireNativeModule } from 'expo';

declare class CellularSignalStrengthModule extends NativeModule{
  signalStrength?: number;
  startListeningToSignalStrength(): void;
  stopListeningToSignalStrength(): void;
}

// This call loads the native module object from the JSI.
export default requireNativeModule<CellularSignalStrengthModule>('CellularSignalStrength');

File: ios/CellularSignalStrengthModule.swift
import ExpoModulesCore

public class CellularSignalStrengthModule: Module {
  public func definition() -> ModuleDefinition {

    Name("CellularSignalStrength")

    Property("signalStrength") {
      return 0
    }

    Function("startListeningToSignalStrength") {}

    Function("stopListenningToSignalStrength") {}

  }
}
